# Laboratorio de regresión - 3

## Significancia de factores

|                |   |
:----------------|---|
| **Nombre**     David Alejandro Rangel Rodríguez|   |
| **Fecha**     29 de enero del 2026 |   |
| **Expediente*756203* |   |

Descarga el archivo de publicidad y carga los datos (Advertising.csv).

In [2]:
import pandas as pd
import numpy as np
import statistics
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

In [3]:
adver = pd.read_csv('Advertising.csv')
adver.head()

Unnamed: 0     TV  radio  newspaper  sales
0           1  230.1   37.8       69.2   22.1
1           2   44.5   39.3       45.1   10.4
2           3   17.2   45.9       69.3    9.3
3           4  151.5   41.3       58.5   18.5
4           5  180.8   10.8       58.4   12.9

**¿Hay alguna relación entre el presupuesto para publicidad y las ventas?**

Nuestra primera meta debe ser determinar si hay evidencia en los datos de que haya una asociación entre estas variables.

- ¿Por qué? ¿Qué resultaría si nos diéramos cuenta de la falta de relación entre el presupuesto de publicidad y las ventas?

Sí, los datos demuestran una relación positiva y muy fuerte entre la inversión publicitaria y las ventas, con un coeficiente de determinación, lo que significa que el presupuesto en publicidad explica un gran porcentaje de la variación ante las ventas.

Si no existiera relación, la conclusión inmediata sería que la estrategia de comunicación actual es ineficaz y que se está desperdiciando capital en canales que no influyen en el consumidor. Esto obligaría a la empresa a detener la inversión publicitaria actual para evitar mayores pérdidas, reevaluar por completo el mensaje o el producto, y desplazar el enfoque hacia otros factores que realmente impulsen el crecimiento,

**¿Qué tan fuerte es esta relación?**
Asumiendo que existe esta relación, ¿nos sirve conocer el impacto que tiene invertir en publicidad en las ventas?

La relación entre la publicidad y las ventas es extremadamente sólida, lo cual es estadísticamente muy significativo y garantiza que los resultados no son casualidad. Conocer este impacto es vital porque otorga capacidad predictiva para estimar ventas futuras, permite la optimización del presupuesto al identificar que medios como la Radio y la TV son mucho más rentables que el Periódico, y facilita el cálculo del retorno de inversión. 

**¿Cuáles medios están asociados con las ventas? ¿Qué tan grande es la asociación entre un medio específico y las ventas?**

Hay 3 medios distintos en los datos. ¿Sirve invertir en los 3? ¿Conviene más invertir sólo en uno?

**¿Qué tan seguros estamos de que podríamos predecir ventas futuras?**

**¿La relación es lineal?**

**¿Hay sinergia entre estos medios?**

Puede ser que gastar \\$50,000 en publicidad y otros \\$50,000 en radio es mejor opción que gastar \\$100,000 en televisión. A esto le llamamos efecto de interacción.

Podemos usar regresión lineal para responder todas estas preguntas.

Realiza una regresión lineal:

$$ \text{ventas} \approx \beta_0 + (\beta_1)(\text{TV})$$

In [4]:
x1 = adver["TV"]
x1 = np.array(x1).reshape(-1, 1)

In [5]:
y1 = adver["sales"]
y1 = np.array(y1).reshape(-1,1)

In [6]:
lr = LinearRegression()
X = np.hstack([x1])
lr.fit(X,y1)
lr.intercept_

array([7.03259355])

In [7]:
lr.coef_

array([[0.04753664]])

### Verificando la precisión de nuestros coeficientes estimados

Recuerda que en el mundo real hay ruidos y errores de medición. Siempre se asume que la verdadera relación entre $X$ y $Y$ es $$Y = \beta_0 + \beta_1 X + \epsilon$$

Se asume que el término de error es independiente de $X$ (el error siempre es el mismo sin importar el valor de $X$). Este modelo describe a la *línea de regresión de la población*, que es la mejor aproximación de la verdadera relación entre $X$ y $Y$. Cuando usamos mínimos cuadrados encontramos la *línea de mínimos cuadrados*.

¿Cuál es la diferencia entre población y muestra?

La población es el conjunto completo y total de todos los elementos o individuos que comparten una característica específica que deseamos estudiar, mientras que la muestra es solo un subgrupo representativo extraído de ese universo para facilitar el análisis. La diferencia clave radica en que trabajar con la población ofrece resultados exactos, pero suele ser costoso o imposible, por lo que utilizamos la muestra para obtener resultados aproximados que luego proyectamos al grupo total mediante la inferencia.

¿Cuál crees que sea la diferencia entre hacer una regresión con todos los datos de la población y una muestra de ella?

La principal diferencia es que la regresión con la población total proporciona los valores reales y exactos de la relación (parámetros), eliminando cualquier incertidumbre, mientras que la regresión con una muestra solo ofrece una estimación (estadísticos) de esa realidad. Debido a que la muestra está sujeta a errores de selección y variabilidad, requiere el uso de herramientas de inferencia como el valor $p$ y los intervalos de confianza para determinar qué tan probable es que los resultados encontrados en ese pequeño grupo realmente se apliquen a todo el universo de datos.

La línea de regresión de la población no se puede observar. El concepto de comparar estas líneas es una extensión natural del acercamiento estadístico estándar de usar información de una muestra para estimar características de una población grande.

Imagina que quieres encontrar la altura promedio de un mexicano $\mu$. Medir a todos y cada uno de los mexicanos en situaciones similares, con la misma regla, mismo operador, y otras incontables formas de minimizar la variación de la medida es una tarea imposible. Lo que podemos asumir es que $\hat{\mu} = \bar{y}$. La media poblacional y la media muestral son diferentes, pero la media muestral es usualmente un buen estimado.

De la misma manera, como no contamos con el 100% de la información para hacer una regresión, los coeficientes $\beta_0$ y $\beta_1$ son desconocidos. Podemos estimarlos usando mínimos cuadrados, encontrando $\hat{\beta_0}$ y $\hat{\beta_1}$. Puede que las muestras que tengamos en ese momento estén un poco por encima de la media, pero otras muestras en otro momento puede que estén debajo de la media. En general, esperamos que el promedio de las aproximaciones $\hat{\mu}$ aproxime a $\mu$.

Esto lleva a la pregunta: ¿qué tan cercanos son nuestros coeficientes estimados a los verdaderos coeficientes? Utilizamos el concepto de error estándar para evaluar esto.

$$ \text{Var}(\hat{\mu})=\text{SE}(\hat{\mu})^2 = \frac{\sigma^2}{n} $$

Donde $\sigma$ es la desviación estándar de cada una de las observaciones $y_i$ de $Y$. El error estándar nos dice la cantidad promedio que el estimado difiere del valor verdadero. Podemos ver en la fórmula que entre más observaciones tengamos el error se hace más pequeño. Las fórmulas para errores estándar de $\hat{\beta_0}$ y $\hat{\beta_1}$ son:

$$ \text{SE}(\hat{\beta_0})^2 = \sigma^2 [\frac{1}{n} + \frac{\bar{x}^2}{\sum_{i=1}^n (x_i - \bar{x})^2}]$$

$$ \text{SE}(\hat{\beta_1})^2 = \frac{\sigma^2}{\sum_{i=1}^n (x_i - \bar{x})^2}$$

$$ \sigma^2 = \text{Var}(\epsilon) = \text{RSE}^2 = \frac{\text{RSS}}{n-p}$$

Para que estas fórmulas sean validas asumimos que los errores $\epsilon_i$ tienen varianza común $\sigma^2$ y que no están correlacionados.

Calcula los errores estándar de los coeficientes

$$ \sigma^2 = \frac{ RSS }{n - p} $$

In [8]:
n,p = X.shape
p = p + 1
n,p

(200, 2)

In [12]:
xbar = np.mean(x1)
aux = np.sum((x1 - xbar)**2)

y_pred = lr.predict(X)
RSS = np.sum((y1 - y_pred)**2) #RSS

var = RSS/(n-p)
var

np.float64(10.618841328946218)

$$ SE(\hat{\beta_1})^2 = \frac{ \sigma^2 }{\sum(x_i-\bar{x})^2} $$

In [13]:
SE_b1 = np.sqrt(var/aux)
SE_b1

np.float64(0.0026906071877968707)

$$ SE(\hat{\beta_0})^2 = \sigma^2 [\frac{1}{n} + \frac{\bar{x}^2}{\sum(x_i-\bar{x})^2}]$$

In [14]:
SE_b0 = np.sqrt(var * (1/n + xbar**2/aux))
SE_b0

np.float64(0.4578429402734785)

Estos errores se pueden usar para calcular intervalos de confianza. Un intervalo de confianza del $95\%$ se define como un rango de valores en el cuál se encuentra el desconocido valor verdadero con un $95\%$ de probabilidad.

Otra forma de verlo es que si tomamos muestras repetidas y construimos un intervalo de confianza para cada una, el $95\%$ de los intervalos creados van a contener el valor verdadero. Para la regresión el intervalo de confianza del $95\%$ toma la forma:

$$ \hat{\beta_j} \pm 2\text{SE}(\hat{\beta_j})$$

Calcula los intervalos de confianza para los coeficientes estimados:

## Intervalo [B_0]

In [15]:
B0 = lr.intercept_
B0
IntermenB0 = B0 - 2*SE_b0
IntermayB0 = B0 + 2*SE_b0
IntervaloB0 = [IntermenB0, IntermayB0]
IntervaloB0

[array([6.11690767]), array([7.94827943])]

## Intervalo [B_1]

In [16]:
B1 = lr.coef_
B1
IntermenB1 = B1 - 2*SE_b1
IntermayB1 = B1 + 2*SE_b1
IntervaloB1 = [IntermenB1,IntermayB1]
IntervaloB1

[array([[0.04215543]]), array([[0.05291785]])]

Los errores estándar también se usan para realizar pruebas de hipótesis. La prueba de hipótesis más común es probar la hipótesis nula de:

$$ H_0: \text{No hay relación entre } X \text{ y } Y \ \ \ \ (\beta_1=0)$$

contra la hipótesis alternativa:
$$ H_0: \text{Hay alguna relación entre } X \text{ y } Y \ \ \ (\beta_1 \neq 0)$$

Explica con tus palabras el significado de la hipótesis nula y la hipótesis alternativa.

La hipótesis nula ($H_0$) es la afirmación que no existe ningún efecto, cambio o relación entre las variables que estás estudiando. Es el punto de partida y siempre se asume como verdadera hasta que los datos demuestren lo contrario. Por el contrario, la hipótesis alternativa ($H_a$) es lo que sospechamos que realmente ocurre: sí existe una relación, diferencia o efecto significativo

Para probal la hipótesis nula debemos determinar si nuestro estimado $\hat{\beta_1}$ de $\beta_1$ está lo suficientemente alejado de cero para que podamos decir con confianza que este valor no es cero. 

¿Qué tan lejos? Depende de qué tanta confianza tengamos en el estimado encontrado. Si nuestro error estándar es pequeño y nuestro estimado está alejado de cero podríamos decir que hay muy poca probabilidad de que el valor verdadero sea 0. En cambio, si nuestro error estándar es grande y nuestro estimado está muy cerca de cero, entonces podrías ser que el valor verdadero sea cero y que no haya relación entre las variables.

Se calcula un *estadístico t* dado por
$$ t = \frac{\hat{\beta_j} - \mu}{\text{SE}(\hat{\beta_j})} $$

donde $\mu$ es el valor contra el que queremos probar.

Calcula el estadístico t para tus coeficientes estimados, usando como referencia la prueba de hipótesis.

#### Estadístico t

In [17]:
mu = 0
tb_0 = (B0 - mu) / SE_b0
tb_0

array([15.36027517])

In [18]:
tb_1 = (B1 - mu) / SE_b1
tb_1

array([[17.6676256]])

La distribución t tiene forma de campana y se parece bastante a la distribución normal cuando $n > 30$. Ya sólo es cuestión de calcular la probabilidad de observar cualquier número tal que su valor absoluto sea igual o mayor que el valor absoluto del estadístico t calculado. En otras palabras:
$$ P(|x| \geq |t|) $$

A esta probabilidad la llamamos *p-value*. Un *p-value* pequeño indica que es poco probable que exista por puro azar una relación significativa entre predictor y respuesta, en caso de que no haya una asociación real entre predictor y respuesta. En otras palabras, el *p-value* te dice la probabilidad de que parezca que hay relación cuando no la hay.

Si el *p-value* es pequeño, inferimos que sí hay una asociación entre el predictor y la respuesta, y **rechazamos la hipótesis nula**.
  

¿Qué tan pequeño? Depende de la aplicación. Un valor muy común es del $5\%$.

Utiliza el siguiente código para calcular el *p-value* para tus coeficientes

`from scipy import stats`

`p_bj = 2*(1 - stats.t.cdf(np.abs(t_bj), n-p))`

In [19]:
from scipy import stats
p_0 = 2*(1 - stats.t.cdf(np.abs(tb_0),n-p))
p_0

array([0.])

In [20]:
p_1 = 2*(1 - stats.t.cdf(np.abs(tb_1),n-p))
p_1

array([[0.]])

¿Se rechaza la hipótesis nula? ¿Qué significa?

Como vemos que valores $p$ obtenidos en el análisis son extremadamente bajos, cercanos a $0$, se rechaza la hipótesis nula. Ya que significa que existe evidencia estadística suficiente para concluir que la inversión en publicidad tiene un efecto real y significativo sobre las ventas, confirmando que "la publicidad sí funciona" para atraer ingresos en este modelo. Por ello si se rechaza la $H_0$.

Realiza otras dos regresiones. Ya tienes hecha la regresión de ventas dado el gasto en publicidad de TV. Realiza la regresión para gastos en radio y gastos en periódico. Organiza las respuestas para que debajo de esta celda se tenga:
- Título de regresión
- Coeficientes estimados
- Errores estándar de los coeficientes
- Intervalos de confianza
- Estadísticos t
- p-values
- Observaciones

## Regresión radio

#### coeficientes estimados

In [46]:
x2 = adver["radio"]
x2 = np.array(x2).reshape(-1, 1)

In [47]:
lr = LinearRegression()
X3 = np.hstack([x2])
lr.fit(X3,y1)
lr.intercept_

array([9.3116381])

In [48]:
lr.coef_

array([[0.20249578]])

#### Errores estántadar de coeficientes

$$ \sigma^2 = \frac{ RSS }{n - p} $$

In [25]:
n,p = X.shape
p = p + 1
n,p

(200, 2)

xbar2 = np.mean(x2)
aux2 = np.sum((x2 - xbar2)**2)

y_pred2 = lr.predict(X3)
RSS2 = np.sum((y1 - y_pred2)**2) #RSS

var2 = RSS2/(n-p)
var2

$$ SE(\hat{\beta_1})^2 = \frac{ \sigma^2 }{\sum(x_i-\bar{x})^2} $$

In [29]:
SE_b1_2 = np.sqrt(var2/aux2)
SE_b1_2

np.float64(0.020411306360090635)

$$ SE(\hat{\beta_0})^2 = \sigma^2 [\frac{1}{n} + \frac{\bar{x}^2}{\sum(x_i-\bar{x})^2}]$$

In [56]:
SE_b0_2 = np.sqrt(var2 * (1/n + xbar2**2/aux2))
SE_b0_2

np.float64(0.5629004962718711)

#### Intervalos de confianza (95%)

**Intervalos Beta_0**

In [33]:
B0_2 = lr.intercept_
B0_2
IntermenB0_2 = B0_2 - 2*SE_b0_2
IntermayB0_2 = B0_2 + 2*SE_b0_2
IntervaloB0_2 = [IntermenB0_2, IntermayB0_2]
IntervaloB0_2

[array([8.1858371]), array([10.43743909])]

**Intervalo Beta_1**

In [60]:
B1_2 = lr.coef_
B1_2
IntermenB1_2 = B1_2 - 2*SE_b1_2
IntermayB1_2 = B1_2 + 2*SE_b1_2
IntervaloB1_2 = [IntermenB1_2,IntermayB1_2]
IntervaloB1_2

[array([[0.01387049]]), array([[0.09551571]])]

#### Estadístico t

$$ t = \frac{\hat{\beta_j} - \mu}{\text{SE}(\hat{\beta_j})} $$

In [63]:
mu = 0
tb_0_2 = (B0_2 - mu) / SE_b0_2
tb_0_2

array([16.54224531])

In [37]:
tb_1_2 = (B1_2 - mu) / SE_b1_2
tb_1_2

array([[9.92076547]])

#### p-values

In [38]:
from scipy import stats
p_0_2 = 2*(1 - stats.t.cdf(np.abs(tb_0_2),n-p))
p_0_2

array([0.])

In [39]:
p_1_2 = 2*(1 - stats.t.cdf(np.abs(tb_1_2),n-p))
p_1_2

array([[0.]])

#### Observaciones

Se observa una relación positiva y bastante rentable, ya que el modelo indica que es el medio con el mayor impacto por cada dólar invertido. Invertir en este canal genera un aumento directo y estadísticamente significativo en las ventas, lo que sugiere que es una herramienta muy eficiente para potenciar los ingresos de forma rápida.

## Regresión periódico

#### coeficientes estimados

In [49]:
x3 = adver["newspaper"]
x3 = np.array(x3).reshape(-1, 1)

In [50]:
lr = LinearRegression()
X4 = np.hstack([x3])
lr.fit(X4,y1)
lr.intercept_

array([12.35140707])

In [51]:
lr.coef_

array([[0.0546931]])

#### Errores estántadar de coeficientes

$$ \sigma^2 = \frac{ RSS }{n - p} $$

In [52]:
n,p = X.shape
p = p + 1
n,p

(200, 2)

In [54]:
xbar3 = np.mean(x3)
aux3 = np.sum((x3 - xbar3)**2)

y_pred3 = lr.predict(X4)
RSS3 = np.sum((y1 - y_pred3)**2) #RSS

var3 = RSS3/(n-p)
var3

np.float64(25.933356283393632)

$$ SE(\hat{\beta_1})^2 = \frac{ \sigma^2 }{\sum(x_i-\bar{x})^2} $$

In [55]:
SE_b1_3 = np.sqrt(var3/aux3)
SE_b1_3

np.float64(0.016575721876358165)

$$ SE(\hat{\beta_0})^2 = \sigma^2 [\frac{1}{n} + \frac{\bar{x}^2}{\sum(x_i-\bar{x})^2}]$$

In [57]:
SE_b0_3 = np.sqrt(var3 * (1/n + xbar3**2/aux3))
SE_b0_3

np.float64(0.6214201876094345)

#### Intervalos de confianza (95%)

**Intervalos Beta_0**

In [59]:
B0_3 = lr.intercept_
B0_3
IntermenB0_3 = B0_3 - 2*SE_b0_3
IntermayB0_3 = B0_3 + 2*SE_b0_3
IntervaloB0_3 = [IntermenB0_3, IntermayB0_3]
IntervaloB0_3

[array([11.10856669]), array([13.59424744])]

**Intervalo Beta_1**

In [61]:
B1_3 = lr.coef_
B1_3
IntermenB1_3 = B1_3 - 2*SE_b1_3
IntermayB1_3 = B1_3 + 2*SE_b1_3
IntervaloB1_3 = [IntermenB1_3,IntermayB1_3]
IntervaloB1_3

[array([[0.02154165]]), array([[0.08784454]])]

#### Estadístico t

$$ t = \frac{\hat{\beta_j} - \mu}{\text{SE}(\hat{\beta_j})} $$

In [65]:
mu = 0
tb_0_3 = (B0_3 - mu) / SE_b0_3
tb_0_3

array([19.87609562])

In [66]:
tb_1_3 = (B1_3 - mu) / SE_b1_3
tb_1_3

array([[3.29959074]])

#### p-values

In [67]:
from scipy import stats
p_0_3 = 2*(1 - stats.t.cdf(np.abs(tb_0_3),n-p))
p_0_3

array([0.])

In [68]:
p_1_3 = 2*(1 - stats.t.cdf(np.abs(tb_1_3),n-p))
p_1_3

array([[0.0011482]])

#### Observaciones

Los datos revelan que su impacto real es prácticamente nulo cuando se analiza junto a los otros medios, ya que su p-value es más alto que los demás. Aunque de forma aislada parece funcionar.

## Regresión lineal múltiple

En lugar de hacer una regresión para cada factor independiente, quizás se puede extender el modelo para que tenga varios factores dentro:

$$ Y = \beta_0 + \beta_1 X_1 + \beta_2 X_2 + ... + \beta_p X_p + \epsilon $$

Para nuestro ejemplo de publicidad:

$$ \text{sales} = \beta_0 + \beta_1 (\text{TV}) + \beta_2 (\text{radio}) + \beta_3 (\text{newspaper}) + \epsilon $$

Utiliza la librería `statsmodels` para realizar la regresión. Por defecto la librería statsmodels no toma en cuenta el intercepto ($\beta_0$), por lo que se tendrá que agregar una columna de unos de tamaño *n* a la matriz X.

`import statsmodels.api as sm`

`ols = sm.OLS(Y, X)`

`results = ols.fit()`

`results.summary()`

In [40]:
matrix = adver[["TV","radio","newspaper"]].to_numpy()
matrix = matrix.reshape(-1,3)
matrix

array([[230.1,  37.8,  69.2],
       [ 44.5,  39.3,  45.1],
       [ 17.2,  45.9,  69.3],
       [151.5,  41.3,  58.5],
       [180.8,  10.8,  58.4],
       [  8.7,  48.9,  75. ],
       [ 57.5,  32.8,  23.5],
       [120.2,  19.6,  11.6],
       [  8.6,   2.1,   1. ],
       [199.8,   2.6,  21.2],
       [ 66.1,   5.8,  24.2],
       [214.7,  24. ,   4. ],
       [ 23.8,  35.1,  65.9],
       [ 97.5,   7.6,   7.2],
       [204.1,  32.9,  46. ],
       [195.4,  47.7,  52.9],
       [ 67.8,  36.6, 114. ],
       [281.4,  39.6,  55.8],
       [ 69.2,  20.5,  18.3],
       [147.3,  23.9,  19.1],
       [218.4,  27.7,  53.4],
       [237.4,   5.1,  23.5],
       [ 13.2,  15.9,  49.6],
       [228.3,  16.9,  26.2],
       [ 62.3,  12.6,  18.3],
       [262.9,   3.5,  19.5],
       [142.9,  29.3,  12.6],
       [240.1,  16.7,  22.9],
       [248.8,  27.1,  22.9],
       [ 70.6,  16. ,  40.8],
       [292.9,  28.3,  43.2],
       [112.9,  17.4,  38.6],
       [ 97.2,   1.5,  30. ],
       [26

In [41]:
import statsmodels.api as sm
X7 = sm.add_constant(matrix)
model = sm.OLS(y1,X7)
results = model.fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.897
Model:                            OLS   Adj. R-squared:                  0.896
Method:                 Least Squares   F-statistic:                     570.3
Date:                Mon, 02 Feb 2026   Prob (F-statistic):           1.58e-96
Time:                        13:25:42   Log-Likelihood:                -386.18
No. Observations:                 200   AIC:                             780.4
Df Residuals:                     196   BIC:                             793.6
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          2.9389      0.312      9.422      0.000       2.324       3.554
x1             0.0458      0.001     32.809      0.000       0.043       0.049
x2             0.1885      0.009     21.893      0.000       0.172       0.206
x3            -0.0010      0.006     -0.177      0.860      -0.013       0.011
==============================================================================
Omnibus:                       60.414   Durbin-Watson:                   2.084
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              151.241
Skew:                          -1.327   Prob(JB):                     1.44e-33
Kurtosis:                       6.332   Cond. No.                         454.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

¿Qué diferencias puedes observar entre los *p-values* de una regresión múltiple y los encontrados en las regresiones simples? ¿Por qué crees que existen estas diferencias?

La principal diferencia es que en las regresiones simples todas las variables parecen tener un efecto significativo, pero en la regresión múltiple el p-value del periódico aumenta drásticamente (de $0.0011$ a $0.8599$), perdiendo su significancia estadística. Esto sucede porque la regresión múltiple evalúa el impacto único de cada variable controlando la presencia de las demás; en este caso, el periódico mostraba una relación aparente en la regresión simple debido a su correlación con la radio, pero una vez que el efecto de la radio se incluye en el modelo, se descubre que el periódico no aporta ningún valor predictivo adicional por sí mismo.

## Referencia

James, G., Witten, D., Hastie, T., Tibshirani, R.,, Taylor, J. (2023). An Introduction to Statistical Learning with Applications in Python. Cham: Springer. ISBN: 978-3-031-38746-3

## Addendum

Para calcular los *p-values* de los parámetros sin `statsmodels`:
1. Calcular RSS
2. Calcular RSE
3. `var_beta = np.linalg.inv(X.T @ X) * rse**2` (X con columna de intercepto)
4. `std_beta = np.sqrt(var_beta.diagonal())` El orden de los valores corresponde al orden de los factores en las columnas de la matriz $X$.
5. Calcular *estadístico t*
6. Calcular *p-value*